src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private void ensureCapacity() {
+        if (size == elements.length) {
+            T[] newArray = (T[]) new Object[(int) (elements.length * 1.5)];
```

Comment:
Magic number '1.5' should be a constant with an informative name to make the code more readable.

Reply:
not fixed